### ISSUE TITLE
ACF Google Map field bug

### ISSUE THREAD

**Issue by Joe-Bloggs** [elliotcondon/acf#788]
Hi Elliot,

I would like to report a possible issue with Google Map field. 

In my case I've got a front-end submission form with Google Map field and a few other required fields. 
When all fields are filled in correctly, the submission goes through with no problems, saving the map location correctly.

But when a required field is left empty, and the page reloads due to the required field error, at this point the map breaks - see the screenshot attached.
![ACF-Google-Map-field-address-bug](https://user-images.githubusercontent.com/33169446/73518382-db411b80-4449-11ea-8ad4-f41c37fee210.jpg)

So I got my developer to take a look. What he did is that in file: 
/plugins/advanced-custom-fields-pro/includes/fields/class-acf-field-google-map.php 

he replaced the following string on line 133:

`$search = $field['value']['address'];`

with the following:

```
if(is_array($field['value'])){
$search = $field['value']['address'];
}
else if(is_string($field['value'])){
$field['value'] = stripslashes($field['value']);
$json_decode_value = json_decode($field['value'],true);
$search = $json_decode_value['address'];
}
```

Is there any chance this could make it into the next update if there is a bug?

Many thanks, JO

**Comment by elliotcondon** [comment]
Thanks @Joe-Bloggs 

Thanks for the bug report and fix!
I'll be more than happy to take a look at this.

There is one detail that has me a little confused: "But when a required field is left empty, and the page reloads due to the required field error, at this point the map breaks - see the screenshot attached."

When a form is submit with an empty required value, one of two things will happen:
1. An AJAX validation request will be made, and JavaScript will display any found errors.
2. The page will reload, but a `wp_die()` function will be called to display the error. This won't allow the page to load any of your template's HTML, and will only display a text message.

In your case, somehow the page is reloading... and the Google Map field is somehow loading a value that is no an array.

Are you able to help me better understand how this is possible? Perhaps there is some custom code running?

**Comment by Joe-Bloggs** [comment]
Hi @elliotcondon 

many thanks for the follow up. 

Yes you might be right. I am using the ACF together with the Directories Pro plugin (https://codecanyon.net/item/directories-pro-for-wordpress/21800540)

The plugin provides front-end submission forms and enables directory listings. On the submission form, I'm allowed to use ACF fields. The fields are already integrated into the template, so I don't need to customise or amend the template to use ACF fields on the form.

When an incomplete form is submitted, it reloads the page, so I guess this is where the problem is. 

Should I be reporting this to the Directories Pro plugin developers instead?

**Comment by elliotcondon** [comment]
Yes, please get in touch with the Directories Pro plugin developers and notify them of the issue. I'm happy to help provide any assistance from my end, but will wait to hear back from you.

**Comment by Joe-Bloggs** [comment]
Sure will do, no worries, thanks Elliot. I'll get back to you with an update shortly.

**Comment by Joe-Bloggs** [comment]
Hey @elliotcondon ,
just a quick update on this. 
The Directories Pro developer released a fix and the issue now seems to be resolved, so all good.
Thanks for your time, 
Jo
